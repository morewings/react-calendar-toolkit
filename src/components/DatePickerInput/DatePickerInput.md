`DatepickerInput` renders provided `Input` with `Datepicker` attached.

```js
import {DatePickerInput} from 'react-calendar-toolkit';

<div>
    <div>Click inside input</div>
    <DatePickerInput onDateSet={date => {console.log('date set', date);}}/>
</div>
```

### `Datepicker` display mode

#### `popover`

Default mode. Position of popover is calculated dynamically to fit in the viewport.

```js
import {DatePickerInput} from 'react-calendar-toolkit';

<div>
    <div>Click inside input</div>
    <DatePickerInput
      mode="popover"
      onDateSet={date => {console.log('date set', date);}}
    />
</div>
```

#### `modal`

Renders `Datepicker` as modal window. More convenient for a small screen devices. 

```js
import {DatePickerInput} from 'react-calendar-toolkit';

<div>
    <div>Click inside input</div>
    <DatePickerInput
      mode="modal"
      onDateSet={date => {console.log('date set', date);}}
    />
</div>
```

### Set `Datepicker` props

Set attached `Datepicker` props via `datePickerProps` property. Same validation rules apply as for `Datepicker` itself, except `onDateSet` callback, `theme` and `dateFnsLocale`, which should be defined in the upper scope.

```js
import locale from 'date-fns/esm/locale/ru';
import {DatePickerInput} from 'react-calendar-toolkit';

const theme = {
  '--bgColor': '#fffff0',
  '--headerBgColor': 'purple',
  '--entrySelectedColor': 'purple',
};

<div>
    <div>Click inside input</div>
    <DatePickerInput
      datePickerProps={{'minPrecision': 'month'}}
      dateFnsLocale={locale}
      theme={theme}
      onDateSet={date => {console.log('date set', date);}}
    />
</div>
```

### `hideOnSelect`

```js
import {DatePickerInput} from 'react-calendar-toolkit';

<div>
    <div>Click inside input</div>
    <DatePickerInput
      hideOnSelect={false}
      onDateSet={date => {console.log('date set', date);}}
    />
</div>
```

### `formatPattern`

Use `date-fns` [format strings](https://date-fns.org/docs/format) to define, what is displayed inside `Input`.

```js
import {DatePickerInput} from 'react-calendar-toolkit';

const theme = {
  '--inputWidth': '160px'
};

<div>
    <div>Click inside input</div>
    <DatePickerInput
      theme={theme}
      formatPattern='GGGG yo Do'
      onDateSet={date => {console.log('date set', date);}}
    />
</div>
```

See [Input custom UI](/#!/Input%20custom%20UI/Input%20custom%20UI) section for more info.