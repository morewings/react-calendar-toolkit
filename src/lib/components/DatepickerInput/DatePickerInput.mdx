---
name: Datepicker attached to the input
route: '/datepicker-input'
---
import { Playground, Props } from 'docz';
import locale from 'date-fns/esm/locale/ru';
import {DatepickerInput} from 'lib';
import DatepickerInputWithProps from './DatepickerInput';

## Description

`DatepickerInput` component renders `Datepicker`, attached to the provided `input` (DOM node or React component). It contains:
- [`Datepicker`](/datepicker) component without the header.
- [`Input`](/src-lib-components-visual-input-input) which is displayed to user.
- [`PopoverWrapper`](/src-lib-components-visual-popover-popover-wrapper) to display `Datepicker` when user clicks (or makes any other action) inside `Input`, when `mode` prop is set to `popover`.
- [`ModalWrapper`](/src-lib-components-visual-modal-modal-wrapper) to display `Datepicker` when user clicks (or makes any other action) inside `Input`, when `mode` prop is set to `modal`.

<Playground>
  <div>
    <div>Click inside input</div>
    <DatepickerInput onDateSet={date => {console.log('date set', date);}}/>
  </div>
</Playground>

## Props

<Props of={DatepickerInputWithProps} />

## Display mode

### `popover`

Position of popover is calculated dynamically to fit inside the viewport.

<Playground>
  <div>
    <div>Click inside input</div>
    <DatepickerInput
      mode="popover"
      onDateSet={date => {console.log('date set', date);}}
    />
  </div>
</Playground>

### `modal`

Renders `Datepicker` as a modal window. More convenient for a small screen devices.

<Playground>
  <div>
    <div>Click inside input</div>
    <DatepickerInput
      mode="modal"
      onDateSet={date => {console.log('date set', date);}}
    />
  </div>
</Playground>

## Set `Datepicker` props

Set attached [`Datepicker` props](/datepicker#props) via `datePickerProps` property. Same validation rules apply as for `Datepicker` itself, except `onDateSet` callback, `theme` and `dateFnsLocale`, which should be defined in the upper scope.

<Playground>
  {() => {
    const theme = {
      '--bgColor': '#fffff0',
      '--headerBgColor': 'purple',
      '--entrySelectedColor': 'purple',
    };
    return (
      <div>
        <div>Click inside input</div>
        <DatepickerInput
          datePickerProps={{'minPrecision': 'month'}}
          dateFnsLocale={locale}
          theme={theme}
          onDateSet={date => {console.log('date set', date);}}
        />
      </div>
    )
  }}
</Playground>


### `hideOnSelect`

Defines `Datepicker` behavior after user clicks on the date entry.

<Playground>
  <div>
    <div>Click inside input</div>
    <DatepickerInput
      hideOnSelect={false}
      onDateSet={date => {console.log('date set', date);}}
    />
  </div>
</Playground>

### `formatPattern`

Use `date-fns` [format strings](https://date-fns.org/docs/format) to define, what is displayed inside `Input`.

<Playground>
  {() => {
    const theme = {
      '--inputWidth': '160px'
    };
    return (
      <div>
        <div>Click inside input</div>
        <DatepickerInput
          theme={theme}
          formatPattern='GGGG yo Do'
          onDateSet={date => {console.log('date set', date);}}
        />
      </div>
    )
  }}
</Playground>
