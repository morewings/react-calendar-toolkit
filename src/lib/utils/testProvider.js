import React from 'react';
import PropTypes from 'prop-types';
import {Provider as LocaleProvider} from 'lib/features/locale/withLocaleContext';
import {Provider as ThemeProvider} from 'lib/features/theme/withTheme';
import {Provider as ModalProvider} from 'lib/features/modal/withModalProvider';
import {Provider as DatepickerProvider} from 'lib/features/datepicker/withDatepickerProvider';

export const DatepickerMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <DatepickerProvider>{props.children}</DatepickerProvider>
    </ThemeProvider>
  </LocaleProvider>
);

export const InputMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <ModalProvider>
        <DatepickerProvider>{props.children}</DatepickerProvider>
      </ModalProvider>
    </ThemeProvider>
  </LocaleProvider>
);

DatepickerMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

InputMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
