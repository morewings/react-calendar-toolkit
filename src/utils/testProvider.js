import React from 'react';
import PropTypes from 'prop-types';
import {Provider as LocaleProvider} from 'features/locale/withLocaleContext';
import {Provider as ThemeProvider} from 'features/theme/withTheme';
import {Provider as ModalProvider} from 'features/modal/withModalProvider';
import {Provider as DatepickerProvider} from 'features/datepicker/withDatepickerProvider';

export const DatepickerMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <DatepickerProvider>{props.children}</DatepickerProvider>
    </ThemeProvider>
  </LocaleProvider>
);

export const InputMockProvider = props => (
  <LocaleProvider {...props}>
    <ThemeProvider>
      <ModalProvider>
        <DatepickerProvider>{props.children}</DatepickerProvider>
      </ModalProvider>
    </ThemeProvider>
  </LocaleProvider>
);

DatepickerMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

InputMockProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
