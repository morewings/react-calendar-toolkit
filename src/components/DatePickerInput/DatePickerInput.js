import React, {useCallback} from 'react';
import PropTypes from 'prop-types';
import {useDispatch, useSelector} from 'react-redux';
import {convertToDate} from 'utils/dateUtils';
import {useFormatDate} from 'utils/localeContext';
import {
  selectors as modalSelectors,
  actionTypes as modalActionTypes,
} from 'features/modal';
import {selectors} from 'features/datepicker';
import DatePicker, {propTypes} from 'components/DatePicker/DatePicker';
import Input from 'components/visual/Input';
import PopoverProvider, {PopoverWrapper} from 'components/visual/Popover';
import ModalProvider, {ModalWrapper} from 'components/visual/Modal';

import useSetInitialValues from 'utils/useSetInitialValues';
import useHasInitialValues from 'utils/useHasInitialValues';

const DatePickerInput = ({
  renderInputAs,
  mode,
  formatPattern,
  renderDatePickerAs,
  initialDate,
  today,
  minPrecision,
  onDateSet,
  popoverProvider,
  wrapPopoverWith,
  modalProvider,
  wrapModalWith,
  wrapInputWith,
  hideOnSelect,
  ...restProps
}) => {
  const dispatch = useDispatch();
  const selectedTimestamp = useSelector(selectors.getSelectedTimestamp);
  const todayTimestamp = useSelector(selectors.getTodayTimestamp);
  const isVisible = useSelector(modalSelectors.getIsVisible);

  const hasInitialValues = useHasInitialValues();

  const handleDateSet = date => {
    onDateSet(date);
    hideOnSelect && toggleDatepicker(false);
  };

  const formatDate = useFormatDate();

  const RenderingLogicProvider =
    mode === 'popover' ? popoverProvider : modalProvider;

  const toggleDatepicker = useCallback(
    visibility =>
      dispatch({
        type: modalActionTypes.TOGGLE_DATEPICKER,
        payload: visibility,
      }),
    [dispatch]
  );

  const InputComponent = renderInputAs;
  const DatePickerComponent = renderDatePickerAs;

  const DatePickerWithProps = () => (
    <DatePickerComponent
      date={convertToDate(selectedTimestamp)}
      showHeader={mode === 'modal'}
      onDateSet={handleDateSet}
      {...restProps}
    />
  );

  useSetInitialValues({initialDate, today, minPrecision});

  return (
    hasInitialValues && (
      <RenderingLogicProvider
        isVisible={isVisible}
        toggleDatepicker={toggleDatepicker}
        wrapPopoverWith={wrapPopoverWith}
        wrapModalWith={wrapModalWith}
        renderDatePickerAs={DatePickerWithProps}>
        <InputComponent
          date={convertToDate(todayTimestamp)}
          value={formatDate(formatPattern, convertToDate(selectedTimestamp))}
          toggleDatepicker={toggleDatepicker}
        />
      </RenderingLogicProvider>
    )
  );
};

DatePickerInput.propTypes = {
  ...propTypes,
  /** Callback when user selects date */
  onDateSet: PropTypes.func.isRequired,
  /** __Rendering__ mode of attached Datepicker */
  mode: PropTypes.oneOf(['popover', 'modal']),
  /** Flag to __show or hide__ Datepicker when date is selected */
  hideOnSelect: PropTypes.bool,
  /** Define component which renders __Input__. */
  renderInputAs: PropTypes.elementType,
  /** Define component which renders __DatePicker__ entry. */
  renderDatePickerAs: PropTypes.elementType,
  /** Define component which wraps __Popover__. */
  wrapPopoverWith: PropTypes.elementType,
  /**
   * Define rendering logic provider for __Popover__.
   * @ignore
   */
  popoverProvider: PropTypes.elementType,
  /**
   * Define rendering logic provider for __Modal__.
   * @ignore
   */
  modalProvider: PropTypes.elementType,
  /** Define component which wraps __Modal__. */
  wrapModalWith: PropTypes.elementType,
  /**
   * date-fns format pattern to format date value inside input
   * @see {@link https://date-fns.org/docs/format|docs}
   */
  formatPattern: PropTypes.string,
  /**
   * Set initial selected date when component renders.
   * @ignore
   */
  initialDate: PropTypes.instanceOf(Date),
  /**
   * Set today date.
   * @ignore
   */
  today: PropTypes.instanceOf(Date),
  /**
   * Set minimum precision (measuring unit) of calendar. Possible values: 'day', 'month', 'year'.
   * @ignore
   */
  minPrecision: PropTypes.oneOf(['year', 'month', 'day']),
};

DatePickerInput.defaultProps = {
  mode: 'popover',
  hideOnSelect: true,
  renderInputAs: Input,
  renderDatePickerAs: DatePicker,
  formatPattern: 'MM/dd/yyyy',
  initialDate: new Date(2020, 0, 6),
  today: new Date(),
  minPrecision: 'day',
  popoverProvider: PopoverProvider,
  wrapPopoverWith: PopoverWrapper,
  wrapModalWith: ModalWrapper,
  modalProvider: ModalProvider,
};

export default DatePickerInput;
