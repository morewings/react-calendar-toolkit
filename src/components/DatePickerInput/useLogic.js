import {useCallback} from 'react';
import {useModalActions} from 'features/modal';
import {convertToDate} from 'utils/dateUtils';
import {useFormatDate} from 'features/locale';
import {useDatePickerContext} from 'features/datepicker';

const useLogic = ({onDateSet, hideOnSelect}) => {
  const {toggleDatePicker} = useModalActions();
  const formatDate = useFormatDate();
  const {
    state: {selectedTimestamp},
  } = useDatePickerContext();

  const handleDateSet = useCallback(
    date => {
      onDateSet(date);
      hideOnSelect && toggleDatePicker(false);
    },
    [hideOnSelect, onDateSet, toggleDatePicker]
  );

  const formatValue = useCallback(
    formatPattern =>
      formatDate(formatPattern, convertToDate(selectedTimestamp)),
    [formatDate, selectedTimestamp]
  );

  const getDate = () => convertToDate(selectedTimestamp);

  return {handleDateSet, formatValue, getDate};
};

export default useLogic;
