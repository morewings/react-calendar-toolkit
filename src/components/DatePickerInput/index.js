import DatePickerInput from './DatePickerInput';

export default DatePickerInput;
