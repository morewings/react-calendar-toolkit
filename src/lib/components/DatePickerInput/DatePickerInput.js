import React from 'react';
import PropTypes from 'prop-types';
import {useModalContext, useModalActions} from 'lib/features/modal';
import {
  useHasInitialValues,
  useSetInitialValues,
} from 'lib/features/datepicker';
import DatePicker, {propTypes} from 'lib/components/DatePicker/DatePicker';
import Input from 'lib/components/visual/Input';
import PopoverProvider, {PopoverWrapper} from 'lib/components/visual/Popover';
import ModalProvider, {ModalWrapper} from 'lib/components/visual/Modal';
import useLogic from './useLogic';

const DatePickerInput = ({
  renderInputAs,
  mode,
  formatPattern,
  renderDatePickerAs,
  onDateSet,
  popoverProvider,
  wrapPopoverWith,
  modalProvider,
  wrapModalWith,
  hideOnSelect,
  datePickerProps,
  datePickerDefaultProps,
}) => {
  const {
    state: {isVisible},
  } = useModalContext();

  const {toggleDatePicker} = useModalActions();

  const {handleDateSet, formatValue, getDate} = useLogic({
    onDateSet,
    hideOnSelect,
  });

  const RenderingLogicProvider =
    mode === 'popover' ? popoverProvider : modalProvider;

  const InputComponent = renderInputAs;
  const DatePickerComponent = renderDatePickerAs;

  const DatePickerWithProps = () => (
    <DatePickerComponent
      showHeader={mode === 'modal'}
      onDateSet={handleDateSet}
      {...datePickerProps}
    />
  );

  const {initialDate, today, minPrecision} = {
    ...datePickerDefaultProps,
    ...datePickerProps,
  };

  useSetInitialValues({initialDate, today, minPrecision});
  const hasInitialValues = useHasInitialValues();

  // eslint-disable-next-line react/prop-types
  const RefWrap = React.forwardRef(({children}, ref) => (
    <div ref={ref}>{children}</div>
  ));

  return (
    hasInitialValues && (
      <RenderingLogicProvider
        isVisible={isVisible}
        toggleDatepicker={toggleDatePicker}
        wrapPopoverWith={wrapPopoverWith}
        wrapModalWith={wrapModalWith}
        renderDatePickerAs={DatePickerWithProps}>
        <RefWrap>
          <InputComponent
            onChange={onDateSet}
            date={getDate()}
            value={formatValue(formatPattern)}
            toggleDatepicker={toggleDatePicker}
          />
        </RefWrap>
      </RenderingLogicProvider>
    )
  );
};

DatePickerInput.propTypes = {
  /** Callback when user selects date */
  onDateSet: PropTypes.func.isRequired,
  /** __Rendering__ mode of attached Datepicker */
  mode: PropTypes.oneOf(['popover', 'modal']),
  /** Flag to __show or hide__ Datepicker when date is selected */
  hideOnSelect: PropTypes.bool,
  /** Define component which renders __Input__. */
  renderInputAs: PropTypes.elementType,
  /**
   * Define component which renders __DatePicker__ entry.
   * @ignore
   */
  renderDatePickerAs: PropTypes.elementType,
  /** Define component which wraps __Popover__. */
  wrapPopoverWith: PropTypes.elementType,
  /**
   * Define rendering logic provider for __Popover__.
   * @ignore
   */
  popoverProvider: PropTypes.elementType,
  /**
   * Define rendering logic provider for __Modal__.
   * @ignore
   */
  modalProvider: PropTypes.elementType,
  /** Define component which wraps __Modal__. */
  wrapModalWith: PropTypes.elementType,
  /**
   * date-fns format pattern to format date value inside input
   * @see https://date-fns.org/docs/format
   */
  formatPattern: PropTypes.string,
  /**
   * Set props for __Datepicker__ component.
   * @see https://morewings.github.io/react-calendar-toolkit/#!/Available%20components/DatePicker
   */
  datePickerProps: PropTypes.shape({
    ...propTypes,
    onDateSet: undefined,
    theme: undefined,
    dateFnsLocale: undefined,
  }),
  /**
   * Default values for __Datepicker__ component.
   * @ignore
   */
  datePickerDefaultProps: PropTypes.shape({
    initialDate: PropTypes.instanceOf(Date),
    today: PropTypes.instanceOf(Date),
    minPrecision: PropTypes.string,
  }),
  /** Theme object to customize style for UI components */
  theme: PropTypes.shape({}), // eslint-disable-line react/require-default-props
  /** date-fns locale object. Defaults to english */
  dateFnsLocale: PropTypes.shape({}), // eslint-disable-line react/require-default-props
};

DatePickerInput.defaultProps = {
  mode: 'popover',
  hideOnSelect: true,
  renderInputAs: Input,
  renderDatePickerAs: DatePicker,
  formatPattern: 'MM/dd/yyyy',
  popoverProvider: PopoverProvider,
  wrapPopoverWith: PopoverWrapper,
  wrapModalWith: ModalWrapper,
  modalProvider: ModalProvider,
  datePickerProps: {},
  datePickerDefaultProps: {
    initialDate: new Date(2020, 0, 8),
    today: new Date(),
    minPrecision: 'day',
  },
};

export default DatePickerInput;
