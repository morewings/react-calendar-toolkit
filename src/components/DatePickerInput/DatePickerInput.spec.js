import React from 'react';
import {render} from '@testing-library/react';
import {InputMockProvider as Provider} from 'utils/testProvider';
import DatePickerInput from './DatePickerInput';

const Datepicker = jest.fn(() => <div>DatePicker</div>);
const Input = jest.fn(() => <div>Input</div>);
const PopoverProvider = jest.fn(({children}) => (
  <div data-testid="Popover">{children}</div>
));
const PopoverWrapper = jest.fn(({children}) => (
  <div data-testid="PopoverWrapper">{children}</div>
));
const ModalProvider = jest.fn(({children}) => (
  <div data-testid="Modal">{children}</div>
));
const ModalWrapper = jest.fn(({children}) => (
  <div data-testid="ModalWrapper">{children}</div>
));

const onDateSet = jest.fn();

const renderWithProps = (props = {}) =>
  render(
    <DatePickerInput
      renderDatePickerAs={Datepicker}
      renderInputAs={Input}
      popoverProvider={PopoverProvider}
      wrapPopoverWith={PopoverWrapper}
      modalProvider={ModalProvider}
      wrapmodalWith={ModalWrapper}
      {...props}
      today={new Date(2020, 0, 10)}
      onDateSet={onDateSet}
    />,
    {
      wrapper: ({children}) => <Provider>{children}</Provider>,
    }
  );

describe('DatePicker', () => {
  beforeEach(() => {
    onDateSet.mockClear();
    Datepicker.mockClear();
    Input.mockClear();
    PopoverProvider.mockClear();
    PopoverWrapper.mockClear();
    ModalProvider.mockClear();
    ModalWrapper.mockClear();
  });

  it('renders with Popover by default', () => {
    const {asFragment} = renderWithProps();
    expect(asFragment()).toMatchSnapshot();
    expect(PopoverProvider.mock.calls[0][0]).toMatchSnapshot();
    expect(Input.mock.calls[0][0]).toMatchSnapshot();
  });

  it('renders with Modal if mode === `modal`', () => {
    const {asFragment} = renderWithProps({mode: 'modal'});
    expect(asFragment()).toMatchSnapshot();
    expect(ModalProvider.mock.calls[0][0]).toMatchSnapshot();
    expect(Input.mock.calls[0][0]).toMatchSnapshot();
  });
});
