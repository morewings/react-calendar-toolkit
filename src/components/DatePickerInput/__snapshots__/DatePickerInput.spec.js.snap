// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders with Modal if mode === \`modal\` 1`] = `
<DocumentFragment>
  <div
    data-testid="Modal"
  >
    <div>
      Input
    </div>
  </div>
</DocumentFragment>
`;

exports[`DatePicker renders with Modal if mode === \`modal\` 2`] = `
Object {
  "children": <mockConstructor
    date={2020-01-08T00:00:00.000Z}
    onChange={[MockFunction]}
    toggleDatepicker={[Function]}
    value="01/08/2020"
  />,
  "isVisible": false,
  "renderDatePickerAs": [Function],
  "toggleDatepicker": [Function],
  "wrapModalWith": [Function],
  "wrapPopoverWith": [MockFunction],
}
`;

exports[`DatePicker renders with Modal if mode === \`modal\` 3`] = `
Object {
  "date": 2020-01-08T00:00:00.000Z,
  "onChange": [MockFunction],
  "toggleDatepicker": [Function],
  "value": "01/08/2020",
}
`;

exports[`DatePicker renders with Popover by default 1`] = `
<DocumentFragment>
  <div
    data-testid="Popover"
  >
    <div>
      Input
    </div>
  </div>
</DocumentFragment>
`;

exports[`DatePicker renders with Popover by default 2`] = `
Object {
  "children": <mockConstructor
    date={2020-01-08T00:00:00.000Z}
    onChange={[MockFunction]}
    toggleDatepicker={[Function]}
    value="01/08/2020"
  />,
  "isVisible": false,
  "renderDatePickerAs": [Function],
  "toggleDatepicker": [Function],
  "wrapModalWith": [Function],
  "wrapPopoverWith": [MockFunction],
}
`;

exports[`DatePicker renders with Popover by default 3`] = `
Object {
  "date": 2020-01-08T00:00:00.000Z,
  "onChange": [MockFunction],
  "toggleDatepicker": [Function],
  "value": "01/08/2020",
}
`;
